ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -129,25 +127,11 @@ public DdlCommandResult executeDropType(final DropTypeCommand dropType) {
           : new DdlCommandResult(true, "Type '" + typeName + "' does not exist");
     }
 
-    private KeyField getKeyField(
-        final Optional<ColumnName> keyFieldName,
-        final LogicalSchema schema
-    ) {
-      if (keyFieldOverride.isPresent()) {
-        return keyFieldOverride.get();
-      }
-      if (keyFieldName.isPresent()) {
-        // for DDL commands, the key name is never specified with a source
-        final ColumnRef columnRef = ColumnRef.withoutSource(keyFieldName.get());
-        final Column keyColumn = schema.findValueColumn(columnRef)
-            .orElseThrow(() -> new IllegalStateException(
-                "The KEY column set in the WITH clause does not exist in the schema: '"
-                    + keyFieldName + "'"
-            ));
-        return KeyField.of(columnRef, keyColumn);
-      } else {
-        return KeyField.none();
-      }
+    private KeyField getKeyField(final Optional<ColumnName> keyFieldName) {
+      return keyFieldOverride
```

Comment:
We should drop `keyFieldOverride` with this change, and just use the key field in the command. It was just being passed in because we needed to preserve the legacy value computed by `SchemaKStream`

Reply:
I did wonder :)

Reply:
I'll pick up in next PR.

Reply:
See https://github.com/confluentinc/ksql/pull/3770